CRM/Utils/Mail.php

```diff
@@ -289,6 +289,14 @@ public static function send(&$params) {
 
     if (is_object($mailer)) {
       $errorScope = CRM_Core_TemporaryErrorScope::ignoreException();
+      // On some servers mail() fails when 'Cc' or 'Bcc' headers are defined but empty.
+      // FIXME: Can we unset Cc/Bcc headers for all mail backends if they are empty?
+      if (empty($headers['Cc'])) {
```

Comment:
@mattwire how about moving this code up i.e. point where it is been initialised
```patch
- $headers['Cc'] = CRM_Utils_Array::value('cc', $params);
- $headers['Bcc'] = CRM_Utils_Array::value('bcc', $params);

+ foreach (['Cc', 'Bcc'] as $fieldPart) {
+  if (!empty($params[$fieldPart])) {
+    $headers[$fieldPart] = $params[$fieldPart];
+  }
+ }
```

and then we no longer this code

Reply:
@monishdeb That makes sense.  The only issue, which @eileenmcnaughton has also mentioned, is how we make this low risk as most servers are happy with the empty headers, but clearly some are not!  Could we end up in a situation where we apply this patch and find that some servers actually *require* the empty headers? :-(

Reply:
right most conservative possible plse

Reply:
@eileenmcnaughton @mattwire I think it's safe to not set these empty attributes to mail headers. Here's more evidence https://www.drupal.org/project/mimemail/issues/960374 https://stackoverflow.com/questions/44884390/laravel-mail-send-even-if-cc-and-bcc-is-null https://github.com/sendgrid/sendgrid-nodejs/issues/263 where they follow the same logic